### DISCUSSION TITLE
PCE with BCS to get Sobol sensitivity

### DISCUSSION THREAD

**Discussion by eclipseyin147** [sandialabs/UQTk#13] [Q&A] [answered]
Hello, I  am new to  UQTk. I am now trying to perform sensitivity analysis with PCE method. The dimension of factors is up to 50. I can not evaluate  samples with size more than the  full expansion count of all PC terms, since it is expensive to simulate them. 
Is it possible to perform BCS to reduce the dimension of PC terms and  get main effects of these factors only by manipulating on  PC coefficients ?  Or I muss resample on the constructed PC surrogate to perform Sobol analysis ?

**Comment by bjdebus** [comment]
Hi @eclipseyin147 , indeed, the scenario you outline is a very common use case for us. As you suggested, if we can not afford to take a lot of samples, we often use BCS to get a PCE that captures the main features in the signal. Once you get that PCE, you can get the Sobol' indices directly from the PCE coefficients either with the app pce_sens or with the function UQTkGSA in https://github.com/sandialabs/UQTk/blob/master/PyUQTk/PyPCE/pce_tools.py

Let me know if you have any further questions or would like an example.

**Reply by eclipseyin147** [reply]
Thank you for your reply. Indeed I tried to perform sensitivity analysis with coefficients from BCs,  namely with pce_sens. I tested it with Sobol function with 8 factors, however the main effect result from pce_sens does not converge to the analytical results.
I passed 300 LHS samples to BCS with PC order 4.  The selected multiindices were passed to construct a PCSet object like below, the "weights" is from WBCS.
`

            /// Compute main sensitivities
           PCSet PCSimpleModel("NISPnoq", newmindex, "LU", 0.0, 1.0);
            Array1D<double> mainsens;
            PCSimpleModel.ComputeMainSens(weights, mainsens);`

**Reply by bjdebus** [reply, marked as answer]
BCS is very powerful, but it also has a lot of hyperparameters that affect the quality of the approximation. Take look at https://github.com/sandialabs/UQTk/blob/master/examples/surrogate_genz/surrogate_genz-BCS.ipynb and https://github.com/sandialabs/UQTk/blob/master/examples/surrogate_genz/surrogate_genz-BCS_detailed.ipynb for some examples. At the very least, you want to try a range of eta values and determine which one minimizes the testing error. Also look at parity plots (predicted vs. true test data) to verify that the surrogate is in the ballpark. If the surrogate is no good, then the Sobol' indices will be off too.
